File: src/api/_fixtures/ai-analysis.js
import { adsList } from './ads'

// ── KPI Summary ────────────────────────────────────────────────────────────────
export const aiSummary = {
  total_analyzed:    856,
  winning_ads:       423,
  winning_ads_trend: 18.2,
  avg_confidence:    74.2,
  avg_conf_trend:    3.1,
  engagement_rate:   3.8,
  engagement_trend:  0.4,
  roas_est:          2.4,
  roas_trend:        0.3,
}

// ── Performance over time (14 days) ───────────────────────────────────────────
const DATES = [
  'Apr 25','Apr 26','Apr 27','Apr 28','Apr 29','Apr 30',
  'May 1','May 2','May 3','May 4','May 5','May 6','May 7','May 8',
]
export const performanceTimeline = DATES.map((date, i) => ({
  date,
  winning_ads:     Math.round(28 + i * 1.3 + Math.sin(i * 0.9) * 4),
  engagement_rate: +Math.max(1.8, 3.1 + Math.sin(i * 0.7) * 0.7 + i * 0.04).toFixed(1),
  roas:            +Math.max(1.2, 2.0 + Math.sin(i * 0.5) * 0.4 + i * 0.025).toFixed(2),
}))

// ── Top angles for donut ───────────────────────────────────────────────────────
export const topAnglesDonut = [
  { name: 'Price/Discount',    value: 2842, pct: 22.1 },
  { name: 'Quality',           value: 2429, pct: 18.9 },
  { name: 'Speed/Convenience', value: 1991, pct: 15.5 },
  { name: 'Social Proof',      value: 1760, pct: 13.7 },
  { name: 'How To/Tutorial',   value: 1246, pct: 9.7  },
  { name: 'Benefits',          value: 1104, pct: 8.6  },
  { name: 'Other',             value: 1470, pct: 11.5 },
]

// ── Confidence score distribution (5 buckets) ─────────────────────────────────
export const confidenceDist = [
  { range: '0–20',   label: 'Very Low', count: 42,  color: '#EF4444' },
  { range: '21–40',  label: 'Low',      count: 118, color: '#F97316' },
  { range: '41–60',  label: 'Medium',   count: 196, color: '#F59E0B' },
  { range: '61–80',  label: 'High',     count: 312, color: '#22C55E' },
  { range: '81–100', label: 'Very High',count: 188, color: '#10B981' },
]

// ── Top winning ads ────────────────────────────────────────────────────────────
const EST_ROAS        = [3.8, 3.2, 2.9, 2.7, 2.5, 2.3, 2.1, 1.9, 1.8, 1.6]
const EST_ENGAGEMENT  = [5.2, 4.8, 4.3, 3.9, 3.7, 3.4, 3.1, 2.9, 2.7, 2.4]

export const winningAds = adsList.data
  .filter((a) => a.status === 'approved')
  .slice(0, 10)
  .map((ad, i) => ({
    ...ad,
    rank:            i + 1,
    est_roas:        EST_ROAS[i] ?? 1.5,
    est_engagement:  EST_ENGAGEMENT[i] ?? 2.0,
  }))

File: backend/app/schemas/ai_analysis.py
"""
AI Analysis page schemas — aligned with frontend fixture (src/api/_fixtures/ai-analysis.js)
and API client (src/api/aiAnalysis.js).

Endpoints served:
  GET /api/ai-analysis/summary
  GET /api/ai-analysis/timeline
  GET /api/ai-analysis/angles
  GET /api/ai-analysis/confidence-dist
  GET /api/ai-analysis/winning-ads
"""

from typing import List, Optional
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict


# ── /summary ──────────────────────────────────────────────────────────────────

class AISummary(BaseModel):
    total_analyzed: int = 0
    winning_ads: int = 0
    winning_ads_trend: float = 0.0       # placeholder — no historical snapshots yet
    avg_confidence: float = 0.0
    avg_conf_trend: float = 0.0          # placeholder
    engagement_rate: float = 0.0         # not available from FB Ad Library — always 0.0
    engagement_trend: float = 0.0        # placeholder
    roas_est: float = 0.0                # not available from FB Ad Library — always 0.0
    roas_trend: float = 0.0              # placeholder


# ── /timeline ─────────────────────────────────────────────────────────────────

class TimelinePoint(BaseModel):
    """One day on the Performance Over Time line chart."""
    date: str                            # e.g. "May 02"
    winning_ads: int = 0
    engagement_rate: float = 0.0         # always 0.0 — no platform metrics
    roas: float = 0.0                    # always 0.0 — no platform metrics


# ── /angles ───────────────────────────────────────────────────────────────────

class AngleDonutItem(BaseModel):
    """One slice of the Top Performing Angles donut chart."""
    name: str
    value: int
    pct: float


# ── /confidence-dist ──────────────────────────────────────────────────────────

class ConfidenceDistItem(BaseModel):
    """One bar in the Confidence Score Distribution bar chart."""
    range: str                           # e.g. "0–20"
    label: str                           # e.g. "Very Low"
    count: int
    color: str                           # hex color matching frontend fixture


# ── /winning-ads ──────────────────────────────────────────────────────────────

class WinningAdCompetitor(BaseModel):
    id: UUID
    name: str
    domain: Optional[str] = None
    logo_url: Optional[str] = None
    tier: int
    region: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class WinningAdInsightField(BaseModel):
    value: Optional[str] = None
    confidence: float = 0.0


class WinningAdAIInsights(BaseModel):
    hook: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    hook_type: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    angle: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    offer_type: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    offer_value: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    creative_format: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    product_line: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    audience_type: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    usp_detected: WinningAdInsightField = Field(default_factory=WinningAdInsightField)
    overall: float = 0.0


class WinningAdResponse(BaseModel):
    """
    Full AdResponse shape + rank + est_roas + est_engagement.
    Matches the fixture: winningAds = adsList.data.map((ad, i) => ({ ...ad, rank, est_roas, est_engagement }))
    """
    # Identity
    id: UUID
    rank: int

    # Competitor ref
    competitor: WinningAdCompetitor

    # Platform + top-level AI fields (denormalized for table speed)
    platform: str
    hook_type: Optional[str] = None
    hook_text: Optional[str] = None
    angle: Optional[str] = None
    angle_detail: Optional[str] = None
    offer_type: Optional[str] = None
    confidence_score: float = 0.0
    status: str

    # Ad content
    headline: Optional[str] = None
    primary_text: Optional[str] = None
    cta: Optional[str] = None
    ad_url: Optional[str] = None
    landing_url: Optional[str] = None
    media_url: Optional[str] = None
    is_video: bool = False
    variants: int = 0

    # Timing
    running_since_days: int = 0
    running_since_date: Optional[str] = None
    captured_at: datetime
    last_seen: datetime
    first_seen: datetime

    notes: Optional[str] = None

    # Full AI insights (nested)
    ai_insights: Optional[WinningAdAIInsights] = None
    ai_notes: Optional[str] = None
    analysis: Optional[dict] = None

    # Estimated metrics — always 0.0 (not available from FB Ad Library)
    est_roas: float = 0.0
    est_engagement: float = 0.0

    # Bookkeeping
    ad_library_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime
